### DISCUSSION TITLE
Should link updates honor preferences over precedence?

### DISCUSSION THREAD

**Discussion by introt** [zim-desktop-wiki/zim-desktop-wiki#1995] [Development]
Currently, automatic link updates try to retain the existing link type. This can result in eg. relative one-part `[[links]]` inside text becoming `[[unintentionally:unwieldy:links]]` after a page move (I encountered this while moving the page containing the links).

The problem can be solved in a number of ways, including

1. Special casing short links – where do we draw the line for the amount of meaningful link parts?
2. Setting the old path as the new link text – may be disorientating and confusing
3. Acting according to the `Prefer short link names for pages` -preference – as mentioned in the title
4. Having the user input `[[these links]]` using Ctrl+L instead of typing the brackets if they don't want the link text to change, and making the Insert Link dialog insert the link text every time as opposed to only when it differs from the target – could be optional per link using another check box.

Somewhat related issues:
#1993

**Comment by introt** [comment]
I'd go for implementing both 3 and 4, as well as exposing the checkbox for `Prefer short link names for pages` in the `Move page` dialog

**Reply by mikro2nd** [reply]
I'd certainly prefer 3. An argument against 4 is that it's another friction/step in the process that is distracting/irritating.
2 could also work for me, but then I avoid hierarchies of pages (aside from Journal pages).
